### PR TITLE
Added flow macros

### PR DESCRIPTION
**Pull request by Tritlo**
I've added flow macros for Hy, both the switch macro, and the guard macro.

### CODE REVIEW

**Comment by Foxboron** [comment]
whats the difference between switch and cond?

**Comment by Tritlo** [comment]
Switch binds to a variable like:

     (switch a
          (= 3 ) (print " a was three")
                    (print "a was not three"))

Though guard is almost exactly like cond, except for the use of [] for each cond, code pair, i.e.

  (cond [(< value 5) (print "value is smaller than 5")]
       [(= value 5) (print "value is equal to 5")]
       [(> value 5) (print "value is greater than 5")]
      [True (print "value is something that it should not be")])

would be

      (guard (< value 5) (print "value is smaller than 5")
                 (= value 5) (print "value is equal to 5")
                 (> value 5) (print "value is greater than 5")
                 True (print "value is something that it should not be"))

**Comment by kirbyfan64** [comment]
IMO, couldn't the macros be in the rST file with all the other macros?

**Comment by nicolas-p** [comment]
Maybe we could have `cond`, `swich` and `case` which all use square brakets to delimit each condition-expression pair, and `cond'`, `swich'` and `case'` in which there are no square brakets and the arguments are implicitely grouped two-by-two.

Square brakets are not necessary in simple examples but when the conditions or expression take several lines, it is good to have them so you can identify when the pairs start and end.

I think both versions are useful.

File: docs/contrib/flow.rst [hunk 1]
```diff
@@ -0,0 +1,60 @@
+==========
+Flow
+==========
+
+.. versionadded:: 0.10.1
+
+The ``flow`` macros allow a programmer to direct the flow of his program with
```
Comment (paultag) [inline docs/contrib/flow.rst]: can we make the `his` gender neuter? :)

**Comment by paultag** [comment]
Very interesting PR series - I like it! LGTM!

**Comment by paultag** [comment]
@hylang/core RFR

**Comment by Foxboron** [comment]
Yepz, LGTM.
:shipit:

**Pull request merged**